client/services.go

```diff
@@ -27,6 +28,9 @@ type ServiceOptions struct {
 	Names []string
 }
 
+// This is used to ensure we send a well-formed service name in the URL path.
+var serviceNameRegexp = regexp.MustCompile(`^[-_A-Za-z0-9]+$`)
```

Comment:
This is an interesting one. The problem I can see where is that services are pebble plan specified (which is way more flexible than a change ID).

Since this is only a client side service name filter, what would happen if someone uses a more exotic service name in his Pebble plan ? 

```
flotter@flotter-pc:~/pebble$ sudo ./pebble services
[sudo] password for flotter: 
Service          Startup  Current  Since
hello-world##@!  enabled  active   today at 21:52 SAST
flotter@flotter-pc:~/pebble$ sudo ./pebble checks
Check            Level  Status  Failures
hello-check##@!  -      up      1/3
flotter@flotter-pc:~/pebble$ sudo ./pebble changes hello-world##@!
ID   Status  Spawn                Ready                Summary
51   Done    today at 21:52 SAST  today at 21:52 SAST  Autostart service "hello-world##@!"
```

My worry is that we are not preventing people from picking exotic service names, but this filter will prevent them accessing the service over the client API ?

Reply:
Yeah, in thinking about this further, I agree with @flotter. We do allow quite the range of service names on the server side, and so we shouldn't restrict that unnecessarily on the client side. It's not actually a security thing, as you point out (they could easily hit the server directly with `curl` and bypass this), so it's possibly breaking some people who choose weird service names, and it's a false sense of security anyway.

Let's keep the check for the change ID (we might as well do it early there, because those have a very well-known format) but not do it for service names. Sorry about the extra work, @IronCore864.